core/data-layer/src/data_layer.rs

```diff
@@ -296,13 +297,23 @@ impl DataLayer {
             return Err(anyhow!(result.err().unwrap().to_string()));
         }
         let result = result.unwrap();
+
+        let config_mapper_data = get_config_mapper();
+
         for row in result {
+            let metadata_str: Option<&str> = row.try_get("metadata").ok();
+            let metadata = if let Some(meta_str) = metadata_str {
```

Comment:
instead of "let = if let", how about "let = match"

Reply:
I modified the statements as follows.
```
let metadata = match row.try_get::<Option<&str>, _>("metadata") {
                Ok(Some(metadata_str)) => {
                    execute_variable_mapper(metadata_str.to_string(), variable_mapper_data.clone())
                        .map_err(|e| anyhow!("Error in execute_variable_mapper: {}", e))?
                }
                Ok(None) => serde_json::Value::Null.to_string(),
                Err(e) => return Err(anyhow!("Error getting metadata: {}", e)),
            };
```